http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -513,6 +530,58 @@ class RuntimeServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
         }
     } yield ()
 
+  private[service] def processDiskConfigRequest(
+    req: DiskConfigRequest,
+    googleProject: GoogleProject,
+    cloudService: CloudService,
+    userInfo: UserInfo
+  )(implicit as: ApplicativeAsk[F, AppContext]): F[PersistentDisk] =
+    (req, cloudService) match {
+      case (DiskConfigRequest.Reference(name), CloudService.GCE) =>
+        for {
+          diskOpt <- persistentDiskQuery.getActiveByName(googleProject, name).transaction
+          disk <- F.fromEither(diskOpt.toRight(DiskNotFoundException(googleProject, name)))
+          attached <- RuntimeServiceDbQueries.isDiskAttachedToRuntime(disk).transaction
+          _ <- if (attached) F.raiseError[Unit](DiskAlreadyAttachedException(googleProject, name)) else F.unit
+          hasAttachPermission <- authProvider.hasPersistentDiskPermission(disk.samResourceId,
```

Comment:
do we need to check if this user is owner/creator for the disk as well?...

Reply:
We could but I don't think it's necessary -- I think Leo should delegate all access control to Sam and not try to make decisions based on owner email in its database.

Reply:
does hasAttach sam call check if the caller(runtime's creator) is owner of the disk as well?

Reply:
This is called in a `create runtime` context.. so the permission check is as follows:
- does the caller have permission to create a runtime in this project?
- does the caller have permission to attach this disk (to any runtime)?

Maybe this permission should be called `attach-on-create` or something… if we supported attaching disks to _existing_ runtimes then yeah we'd probably need some additional permission checks.

Reply:
okay...so I guess we're just saying disks can be attached to anyone's runtime is by design then. Would be nice to mention this in some doc. (I think @akarukappadath might be working on a doc around sam permissions for disks, would be nice to add this there)